shotover/src/message/mod.rs

```diff
@@ -64,6 +64,9 @@ impl From<&ProtocolType> for CodecState {
 
 pub type Messages = Vec<Message>;
 
+/// Unique identifier for the message assigned by shotover at creation time.
+pub type MessageId = u128;
```

Comment:
Why 128bits? This means 2 registers, for every single message. Can't this run on 32? 4_294_967_295 Messages before we have to rollover?

Reply:
Because it's random, not incrementing. I think you would get collisions on a u32.

I didn't measure any performance issues with u128, so I didn't consider it worth investigating at the moment.
Further down the line I could do some micro benchmarks with a global incrementing u32.

Reply:
Why not incrementing?

Reply:
Atomic contention.
I'll investigate an incrementing approach in a follow up.

Reply:
It's possible we could just have each message only be unique to it's chain. They'd still be unique across Shotover with the chain name and id number to differentiate.